### PR TITLE
Add font version_major and version_minor options

### PR DESCRIPTION
**Pull request by yisibl**
https://docs.microsoft.com/en-us/typography/opentype/otspec181/name#name-ids:~:text=Version%20string

> Version string. Should begin with the syntax ‘Version <number>.<number>’ (upper case, lower case, or mixed, with a space between “Version” and the number).
>
>The string must contain a version number of the following form: one or more digits (0-9) of value less than 65,535, followed by a period, followed by one or more digits of value less than 65,535. Any character other than a digit will terminate the minor number. A character such as “;” is helpful to separate different pieces of version information.
>
>The first such match in the string can be used by installation software to compare font versions. Note that some installers may require the string to start with “Version ”, followed by a version number as above.

### CODE REVIEW

**Review by anthrotype — COMMENTED** [review]
this looks good. Could you add a test as well?

File: tests/write_font_test.py [hunk 1]
```diff
@@ -41,6 +41,34 @@ def test_keep_glyph_names(svgs, color_format, keep_glyph_names):
         assert ttfont["post"].formatType == 3.0
         assert ufo.glyphOrder != ttfont.getGlyphOrder()
 
+@pytest.mark.parametrize(
+    "svgs", [("rect.svg", "one-o-clock.svg")]
+)
+@pytest.mark.parametrize(
+    "color_format", ["glyf", "cff2_colr_0", "cff2_colr_1", "cff_colr_0", "cff_colr_1", "glyf_colr_0", "glyf_colr_1", "picosvg", "untouchedsvg"]
```
Comment (yisibl) [inline tests/write_font_test.py:48]: Do we need to test all font formats?

Comment (anthrotype) [reply tests/write_font_test.py]: i don't think so, the version setting is orthogonal to the color font format and applies to all of them indistinctly as they all will have a head table and a name table, like any opentype fonts. I think you can just test a couple, like glyf_colr_1 and cff_colr_1

Comment (yisibl) [reply tests/write_font_test.py]: Done, PTAL.

File: src/nanoemoji/config.py [hunk 2]
```diff
@@ -47,8 +47,10 @@
 
 # we use None as a sentinel for flag not set; FontConfig class has the actual defaults.
 # CLI flags override config file (which overrides default FontConfig).
-flags.DEFINE_string("config", None, "Config file")
+flags.DEFINE_string("config", None, "Config file.")
 flags.DEFINE_integer("upem", None, "Units per em.")
+flags.DEFINE_integer("version_major", None, "Major version.")
```
Comment (anthrotype) [inline src/nanoemoji/config.py:52]:
```suggestion
flags.DEFINE_integer("version_major", 1, "Major version.")
```

File: src/nanoemoji/config.py [hunk 3]
```diff
@@ -47,8 +47,10 @@
...
+flags.DEFINE_integer("version_minor", None, "Minor version.")
```
Comment (anthrotype) [inline src/nanoemoji/config.py:53]:
```suggestion
flags.DEFINE_integer("version_minor", 0, "Minor version.")
```

File: tests/test_helper.py [hunk 4]
```diff
@@ -59,6 +59,7 @@ def color_font_config(config_overrides, svgs, tmp_dir=None):
             upem=100,
             keep_glyph_names=True,
             fea_file=fea_file,
+            # version_minor=2,
```
Comment (anthrotype) [inline tests/test_helper.py:62]: what's this?

File: tests/write_font_test.py [hunk 5]
```diff
@@ -42,6 +42,40 @@ def test_keep_glyph_names(svgs, color_format, keep_glyph_names):
...
+        "cff_colr_1",
+        "glyf_colr_0",
+        "glyf_colr_1",
+        "picosvg",
+    ],
+)
+@pytest.mark.parametrize("version_major", [12, 9])
+@pytest.mark.parametrize("version_minor", [0, "008"])
+# Test version 16.28
+def test_version_minor(svgs, color_format, version_major, version_minor):
```
Comment (anthrotype) [inline tests/write_font_test.py:59]: why do you parametrize the `version_major ` and `version_minor` if you then discard the parametrized value and hard-code 16.28 in the body of the test function?

File: tests/write_font_test.py [hunk 6]
```diff
@@ -42,6 +42,40 @@ def test_keep_glyph_names(svgs, color_format, keep_glyph_names):
...
+            "version_minor": 28,
+        },
+        svgs,
+    )
+    ufo, ttfont = write_font._generate_color_font(config, glyph_inputs)
+    ttfont = test_helper.reload_font(ttfont)
+
+    if version_major:
+        assert ufo.info.versionMajor == 16
+    elif version_minor:
```
Comment (anthrotype) [inline tests/write_font_test.py:73]: this `if ... elif ...` block does not make any sense. why don't just assert the value is as expected, why if?

File: tests/write_font_test.py [hunk 7]
```diff
@@ -42,6 +42,40 @@ def test_keep_glyph_names(svgs, color_format, keep_glyph_names):
...
+        assert ufo.info.versionMinor == 28
+    # else:
+    # TODO: test default version 1.000
```
Comment (anthrotype) [inline tests/write_font_test.py:76]: you may as well add a separate test for that

**Review by anthrotype — COMMENTED** [review]
I left some comments.

In the tests you want to check that the version string (nameID=5) in the name table is as expected. E.g.

```python
assert "Version 16.28" in ttfont["name"].getDebugName(5)
```

File: src/nanoemoji/config.py (hunk 2 above)
Comment (anthrotype) [reply src/nanoemoji/config.py:52]: Actually, it's correct like this. The flags default should be all `None`, in this way the FontConfig default values will be used when a config parameter is not overridden in a config file nor with a CLI flag. Am I right @rsheeter?

Comment (rsheeter) [reply src/nanoemoji/config.py:52]: Correct, we take None as "no flag provided"; the default goes on the Config struct

File: tests/test_helper.py (hunk 4 above)
Comment (yisibl) [reply tests/test_helper.py]: Removed.

File: tests/write_font_test.py (hunk 5 above)
Comment (yisibl) [reply tests/write_font_test.py]: Done

File: tests/write_font_test.py (hunk 6 above)
Comment (yisibl) [reply tests/write_font_test.py]: Done

File: tests/write_font_test.py (hunk 7 above)
Comment (yisibl) [reply tests/write_font_test.py]: Add `test_default_version`

**Comment by yisibl** [comment]
> I left some comments.
> 
> In the tests you want to check that the version string (nameID=5) in the name table is as expected. E.g.
> 
> ```python
> assert "Version 16.28" in ttfont["name"].getDebugName(5)
> ```

Done.

Add openTypeNameVersion and openTypeNameUniqueID test case

**Comment by anthrotype** [comment]
sorry, I merged before you finished, I thought you were done. Anyway please have a look at https://github.com/googlefonts/nanoemoji/pull/253

**Pull request merged**